`jbang info classpath` raises `java.lang.NullPointerException`

**Describe the bug**
```bash
jbang info classpath --verbose --catalog=jbang-catalog.json datahub-protobuf
```

raises `java.lang.NullPointerException` when using this catalog:

```json
{
  "catalogs": {},
  "aliases": {
    "datahub-protobuf": {
      "script-ref": "https://raw.githubusercontent.com/datahub-project/datahub/b9068ffd2ef527a2b24b5d3828a9cc6c0f23b4fb/metadata-integration/java/datahub-protobuf-example/libs/datahub-protobuf.jar",
      "dependencies": [
        "org.slf4j:slf4j-simple:1.7.30",
        "io.acryl:datahub-client:0.8.44",
        "com.google.protobuf:protobuf-java:3.19.3",
        "org.jgrapht:jgrapht-core:1.5.1",
        "com.google.guava:guava:27.0.1-jre",
        "com.google.code.gson:gson:2.8.6"
      ],
      "main": "datahub.protobuf.App"
    }
  },
  "templates": {}
}
```

```console
$  jbang info classpath --verbose --catalog=jbang-catalog.json datahub-protobuf
[jbang] jbang version 0.98.0
[jbang] Reading catalog from /srv/workspace/jbang-catalog.json
[jbang] Duplicating classpath resource jbang-catalog.json
[jbang] Reading catalog from /jbang/.jbang/cache/urls/2c97f3a787487bf73b7f4d02e109487e0e43be594ef95072827efd1500cd99ae
[jbang] Downloaded file https://raw.githubusercontent.com/datahub-project/datahub/b9068ffd2ef527a2b24b5d3828a9cc6c0f23b4fb/metadata-integration/java/datahub-protobuf-example/libs/datahub-protobuf.jar
[jbang] [ERROR] public class java.lang.NullPointerException
java.lang.NullPointerException
	at dev.jbang.cli.BaseInfoCommand$ScriptInfo.init(Info.java:118)
	at dev.jbang.cli.BaseInfoCommand$ScriptInfo.<init>(Info.java:84)
	at dev.jbang.cli.BaseInfoCommand.getInfo(Info.java:162)
	at dev.jbang.cli.ClassPath.doCall(Info.java:198)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:136)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:20)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at dev.jbang.cli.JBang$3.handle(JBang.java:151)
	at dev.jbang.cli.JBang$3.handle(JBang.java:146)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
[jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues

```

**To Reproduce**
Steps to reproduce the behavior:

1. Save the following in `jbang-catalog.json`:
```json
{
  "catalogs": {},
  "aliases": {
    "datahub-protobuf": {
      "script-ref": "https://raw.githubusercontent.com/datahub-project/datahub/b9068ffd2ef527a2b24b5d3828a9cc6c0f23b4fb/metadata-integration/java/datahub-protobuf-example/libs/datahub-protobuf.jar",
      "dependencies": [
        "org.slf4j:slf4j-simple:1.7.30",
        "io.acryl:datahub-client:0.8.44",
        "com.google.protobuf:protobuf-java:3.19.3",
        "org.jgrapht:jgrapht-core:1.5.1",
        "com.google.guava:guava:27.0.1-jre",
        "com.google.code.gson:gson:2.8.6"
      ],
      "main": "datahub.protobuf.App"
    }
  },
  "templates": {}
}
```
2. Run `jbang info classpath --verbose --catalog=jbang-catalog.json datahub-protobuf`

**Expected behavior**
I expect to see the classpath that will be used with run.

**JBang version**
```console
$ jbang version --verbose
[jbang] jbang version 0.98.0
Cache: /jbang/.jbang/cache
Config: /jbang/.jbang
Repository:/root/.m2/repository
0.98.0

```

**Additional context**

Can reproduce with this docker compose environment: https://github.com/aucampia/issues/tree/5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf


```bash
curl --silent -L https://github.com/aucampia/issues/archive/5d96bbb06d0581a233e0a0c41198362770a48af9.tar.gz \
  | tar --strip-components 1 --wildcards -zxvf - -C /var/tmp/ '*/20220908-datahub-protobuf/'
(cd /var/tmp/20220908-datahub-protobuf/ \
  && docker compose build \
  && docker compose run --rm devtools \
    jbang info classpath --verbose --catalog=jbang-catalog.json datahub-protobuf)
```

```
20220908T172144 iwana@iwana-pc00.coop.no:~
$ curl --silent -L https://github.com/aucampia/issues/archive/5d96bbb06d0581a233e0a0c41198362770a48af9.tar.gz \
>   | tar --strip-components 1 --wildcards -zxvf - -C /var/tmp/ '*/20220908-datahub-protobuf/'
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/.gitignore
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/Makefile
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/README.md
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/buf.gen.yaml
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/buf.work.yaml
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/devtools/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/devtools/Dockerfile
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/docker-compose.yaml
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/jbang-catalog.json
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/buf.lock
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/buf.yaml
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/example/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/example/greeter/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/example/greeter/v1/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/example/greeter/v1/greeter.proto
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/io/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/io/cloudevents/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/io/cloudevents/v1/
issues-5d96bbb06d0581a233e0a0c41198362770a48af9/20220908-datahub-protobuf/proto/io/cloudevents/v1/cloudevents.proto
20220908T172145 iwana@iwana-pc00.coop.no:~
$ (cd /var/tmp/20220908-datahub-protobuf/ \
>   && docker compose build \
>   && docker compose run --rm devtools \
>     jbang info classpath --verbose --catalog=jbang-catalog.json datahub-protobuf)
Sending build context to Docker daemon     650B
Step 1/10 : FROM docker.io/mikefarah/yq:4@sha256:495c1e1db2d653cce61a06da52cfca0c7d68d6249cc6e61b2a134d92c609c016 AS yq
 ---> 9e414b60bebc
Step 2/10 : FROM docker.io/bufbuild/buf:1.7.0@sha256:21aaf6aae139b668fb23daeaef5b2b2172893a8a7d9071c8fd965cd7dd9764b1 AS buf
 ---> e961e9297b35
Step 3/10 : FROM docker.io/jbangdev/jbang-action:0.98.0@sha256:d6c55f555177d226dcd3e85d9da6fe95833a9d102438d75ed8ea12273eb329a7 AS jbang-action
 ---> 570492a0ed79
Step 4/10 : FROM jbang-action AS base
 ---> 570492a0ed79
Step 5/10 : SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ---> Using cache
 ---> 512e87828669
Step 6/10 : RUN     apt-get update &&     apt-get install -y         --no-install-recommends         protobuf-compiler         make         socat         ca-certificates         curl         busybox         git         rsync         libarchive-tools         &&     rm -vr /var/lib/apt/lists/* &&     true
 ---> Using cache
 ---> 69ab463b2fab
Step 7/10 : COPY --from=yq /usr/bin/yq /usr/local/bin/
 ---> Using cache
 ---> 5c521cbf00b6
Step 8/10 : COPY --from=buf /usr/local/bin/buf /usr/local/bin/
 ---> Using cache
 ---> d11991b6bbb6
Step 9/10 : RUN     jbang trust add https://github.com/datahub-project/datahub/ &&     jbang trust add  https://raw.githubusercontent.com/datahub-project/datahub/ &&     true
 ---> Using cache
 ---> ca4e73f6485a
Step 10/10 : ENTRYPOINT []
 ---> Running in 30958f443b0c
Removing intermediate container 30958f443b0c
 ---> f7ede5b81822
Successfully built f7ede5b81822
Successfully tagged 20220908-datahub-protobuf_devtools:latest
[jbang] jbang version 0.98.0
[jbang] Reading catalog from /srv/workspace/jbang-catalog.json
[jbang] Duplicating classpath resource jbang-catalog.json
[jbang] Reading catalog from /jbang/.jbang/cache/urls/2c97f3a787487bf73b7f4d02e109487e0e43be594ef95072827efd1500cd99ae
[jbang] Downloaded file https://raw.githubusercontent.com/datahub-project/datahub/b9068ffd2ef527a2b24b5d3828a9cc6c0f23b4fb/metadata-integration/java/datahub-protobuf-example/libs/datahub-protobuf.jar
[jbang] [ERROR] public class java.lang.NullPointerException
java.lang.NullPointerException
	at dev.jbang.cli.BaseInfoCommand$ScriptInfo.init(Info.java:118)
	at dev.jbang.cli.BaseInfoCommand$ScriptInfo.<init>(Info.java:84)
	at dev.jbang.cli.BaseInfoCommand.getInfo(Info.java:162)
	at dev.jbang.cli.ClassPath.doCall(Info.java:198)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:136)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:20)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
	at dev.jbang.cli.JBang$3.handle(JBang.java:151)
	at dev.jbang.cli.JBang$3.handle(JBang.java:146)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
[jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

--- a/src/main/java/dev/jbang/cli/Info.java
+++ b/src/main/java/dev/jbang/cli/Info.java
@@ -80,8 +80,7 @@ abstract class BaseInfoCommand extends BaseCommand {
 			if (scripts.add(originalResource)) {
 				backingResource = code.getResourceRef().getFile().toString();
 
-				Source source = code.asProject().getMainSource();
-				init(source);
+				init(code.asProject());
 
 				if (ctx != null) {
 					applicationJar = code.getJarFile() == null ? null : code.getJarFile().toAbsolutePath().toString();
@@ -112,6 +111,25 @@ abstract class BaseInfoCommand extends BaseCommand {
 			originalResource = source.getResourceRef().getOriginalResource();
 			backingResource = source.getResourceRef().getFile().toString();
 			init(source);
+		}
+
+		private void init(Project prj) {
+			List<String> deps = prj.resolveClassPath().getClassPaths();
+			if (!deps.isEmpty()) {
+				dependencies = deps;
+			}
+			if (prj.getMainSource() == null) {
+				if (!prj.getRepositories().isEmpty()) {
+					repositories = prj	.getRepositories()
+										.stream()
+										.map(Repo::new)
+										.collect(Collectors.toList());
+				}
+				gav = prj.getGav().orElse(null);
+				description = prj.getDescription().orElse(null);
+			} else {
+				init(prj.getMainSource());
+			}
 		}
 
 		private void init(Source source) {
@@ -171,6 +189,7 @@ abstract class BaseInfoCommand extends BaseCommand {
 		ctx.setAdditionalSources(scriptMixin.sources);
 		ctx.setAdditionalResources(scriptMixin.resources);
 		ctx.setForceType(scriptMixin.forceType);
+		ctx.setCatalog(scriptMixin.catalog);
 		return ctx;
 	}
 

--- a/src/main/java/dev/jbang/source/Project.java
+++ b/src/main/java/dev/jbang/source/Project.java
@@ -152,7 +152,7 @@ public class Project implements Code {
 		return this;
 	}
 
-	@Nullable
+	@Nonnull
 	public Optional<String> getDescription() {
 		return Optional.ofNullable(description);
 	}
@@ -163,7 +163,7 @@ public class Project implements Code {
 		return this;
 	}
 
-	@Nullable
+	@Nonnull
 	public Optional<String> getGav() {
 		return Optional.ofNullable(gav);
 	}
@@ -212,7 +212,7 @@ public class Project implements Code {
 		return getMainSourceSet().updateDependencyResolver(resolver);
 	}
 
-	@Nonnull
+	@Nullable
 	public Source getMainSource() {
 		return mainSource;
 	}

--- a/src/test/java/dev/jbang/cli/TestInfo.java
+++ b/src/test/java/dev/jbang/cli/TestInfo.java
@@ -1,16 +1,7 @@
 package dev.jbang.cli;
 
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.allOf;
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.endsWith;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.everyItem;
-import static org.hamcrest.Matchers.greaterThan;
-import static org.hamcrest.Matchers.hasItem;
-import static org.hamcrest.Matchers.hasSize;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.*;
 
 import java.nio.file.Paths;
 import java.util.Collection;
@@ -146,5 +137,19 @@ public class TestInfo extends BaseTest {
 		assertThat(info.resolvedDependencies, Matchers.<Collection<String>>allOf(
 				hasSize(equalTo(1)),
 				everyItem(containsString("commons-lang3"))));
+	}
+
+	@Test
+	void testInfoHelloJar() {
+		String jar = examplesTestFolder.resolve("hellojar.jar").toString();
+		CommandLine.ParseResult pr = JBang.getCommandLine().parseArgs("info", "tools", jar);
+		Tools tools = (Tools) pr.subcommand().subcommand().commandSpec().userObject();
+		BaseInfoCommand.ScriptInfo info = tools.getInfo();
+		assertThat(info.originalResource, equalTo(jar));
+		assertThat(info.applicationJar, equalTo(jar));
+		assertThat(info.backingResource, equalTo(jar));
+		assertThat(info.javaVersion, not(nullValue()));
+		assertThat(info.mainClass, equalTo("helloworld"));
+		assertThat(info.resolvedDependencies, empty());
 	}
 }

